src/LayerEditor/ui/panels/surfaces.py

```diff
@@ -202,45 +99,38 @@ def set_icon(self):
             self.setIcon(icon.get_app_icon("hidden"))
 
 
-
-
-
-
-"""TODO: must set a current surface, need handler in main class."""
-
 class Surfaces(QtWidgets.QWidget):
     """
     GeoMop Layer editor surfaces panel
     
     pyqtSignals:
         * :py:attr:`show_grid
-
-    All regions function contains history operation without label and
-    must be placed after first history operation with label.
     """
-
-
     show_grid = QtCore.pyqtSignal(bool)
-    """Signal is sent when mash shoud be show o repaint.    
+    """Signal is sent when mash should be shown or repaint.    
     :param bool force: if force not set, don't call mash if already exist
     """
+    first_surface_added = QtCore.pyqtSignal(QtCore.QRectF)
+    """ Signal for changing init area rect based on first loaded surface.
+        :param QRectF init_area: area of the first surface
+    """
 
-
-    def __init__(self, layers, home_dir, parent=None):
+    def __init__(self, le_model, main_window, parent=None):
         """
-        :param layers: Layers data object (i.e. parent data object where we may read and write the data)
-        :param home_dir: Where to start Open file dialog.
-                        TODO: introduce a sort of file management object, that can provide this info and save recent ...
+        :param le_model: LEModel data object (i.e. parent data object where we may read and write the data)
         :param parent: Surface panel parent.
         """
         super().__init__(parent)
-
+        self.mainwindow = main_window
```

Comment:
Why passing the whole le_model and main_window in?

Reply:
Because le_model holds many neccesary variables and methods which SurfacePanel needs. main_window doesnt need to be passed, I passed only save_file method which is needed.

Reply:
For almost all the surfaces_model should be enough. Filename can be passed as a string.